internal/build/phase_test.go

```diff
@@ -261,13 +264,63 @@ func testPhase(t *testing.T, when spec.G, it spec.S) {
 
 			when("#WithDaemonAccess", func() {
 				it("allows daemon access inside the container", func() {
-					configProvider := build.NewPhaseConfigProvider(phaseName, lifecycleExec, build.WithArgs("daemon"), build.WithDaemonAccess())
+					configProvider := build.NewPhaseConfigProvider(phaseName, lifecycleExec, build.WithArgs("daemon"), build.WithDaemonAccess(""))
 					phase := phaseFactory.New(configProvider)
 					assertRunSucceeds(t, phase, &outBuf, &errBuf)
 					h.AssertContains(t, outBuf.String(), "daemon test")
 				})
 			})
 
+			when("#WithDaemonAccess with non-standard mount", func() {
+				it.Before(func() {
+					h.SkipIf(t, runtime.GOOS != "linux", "Skipped on non-linux")
+				})
+				it("allows daemon access inside the container", func() {
+					tmp, err := ioutil.TempDir("", "testSocketDir")
+					if err != nil {
+						t.Fatal(err)
+					}
+					defer os.RemoveAll(tmp)
+
+					newName := filepath.Join(tmp, "docker.sock")
+					err = os.Symlink("/var/run/docker.sock", newName)
+					if err != nil {
+						t.Fatal(err)
+					}
+
+					configProvider := build.NewPhaseConfigProvider(phaseName, lifecycleExec, build.WithArgs("daemon"), build.WithDaemonAccess("unix://"+newName))
+					phase := phaseFactory.New(configProvider)
+					assertRunSucceeds(t, phase, &outBuf, &errBuf)
+					h.AssertContains(t, outBuf.String(), "daemon test")
+				})
+			})
+
+			when("#WithDaemonAccess with TCP docker-host", func() {
+				it.Before(func() {
+					h.SkipIf(t, runtime.GOOS != "linux", "Skipped on non-linux")
+				})
+				it("allows daemon access inside the container", func() {
+					forwardCtx, cancelForward := context.WithCancel(context.Background())
+					defer cancelForward()
+
+					forwardExited := make(chan struct{}, 1)
+					go func() {
+						forwardUnix2TCP(forwardCtx, t)
+						forwardExited <- struct{}{}
+					}()
+					time.Sleep(time.Second * 1) // just to make sure `forward` is up and running
+					configProvider := build.NewPhaseConfigProvider(phaseName, lifecycleExec,
+						build.WithArgs("daemon"),
+						build.WithDaemonAccess("tcp://127.0.0.1:12345"),
+						build.WithNetwork("host"))
+					phase := phaseFactory.New(configProvider)
+					assertRunSucceeds(t, phase, &outBuf, &errBuf)
+					h.AssertContains(t, outBuf.String(), "daemon test")
+					cancelForward()
+					<-forwardExited
+				})
+			})
```

Comment:
nit: Move these inside the #WithDaemonAccess `when` block above it, so that the tests are nested